resources/rebuild.sh

```diff
@@ -199,31 +201,121 @@ function build_al_kernel {
-compile_and_install $BIN/init.c    $BIN/init
-compile_and_install $BIN/fillmem.c $BIN/fillmem
-compile_and_install $BIN/fast_page_fault_helper.c $BIN/fast_page_fault_helper
-compile_and_install $BIN/readmem.c $BIN/readmem
-if [ $ARCH == "aarch64" ]; then
-    compile_and_install $BIN/devmemread.c $BIN/devmemread
-fi
+    # Create the directory in which we will store the kernels
+    mkdir -p $OUTPUT_DIR
 
-build_rootfs ubuntu-22.04 jammy
-build_initramfs
+    if [[ "$KERNEL_VERSION" == @(all|5.10) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10.config
+    fi
+    if [[ $ARCH == "x86_64" && "$KERNEL_VERSION" == @(all|5.10-no-acpi) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10-no-acpi.config
+    fi
+    if [[ "$KERNEL_VERSION" == @(all|6.1) ]]; then
+        build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-6.1.config 5.10
+    fi
+}
 
-clone_amazon_linux_repo
+function build_non_ci_kernel {
+    local LINUX_REPO_URL=$1
+    local LINUX_KCONFIG_PATH=$2
 
-# Apply kernel patches on top of AL configuration
-apply_kernel_patches_for_ci
+    clone_linux_repo $LINUX_REPO_URL
+    build_al_kernel $LINUX_KCONFIG_PATH
+}
 
-build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10.config
-if [ $ARCH == "x86_64" ]; then
-    build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-5.10-no-acpi.config
-fi
-build_al_kernel $PWD/guest_configs/microvm-kernel-ci-$ARCH-6.1.config
+function print_help {
+    echo ""
+    echo "Firecracker CI artifacts build script"
+    echo "Usage: $(basename $0) [<command>] [<command args>]"
+    echo ""
+    echo "Available commands:"
+    echo ""
+    echo "    all (default)"
+    echo "        Build CI rootfs and default guest kernels using configurations from"
+    echo "        resources/guest_configs."
+    echo "        This will patch the guest configurations with all the patches under"
+    echo "        resources/guest_configs/patches."
+    echo "        This is the default command, if no command is chosen."
+    echo ""
+    echo "    rootfs"
+    echo "        Builds only the CI rootfs."
+    echo ""
+    echo "    kernels [version]"
+    echo "        Builds our the currently supported CI kernels."
+    echo ""
+    echo "        version: Optionally choose a kernel version to build. Supported"
+    echo "                 versions are: 5.10, 5.10-no-acpi or 6.1."
+    echo ""
+    echo "    help"
+    echo "        Displays the help message and exits."
+    
+}
+
+function main {
+    if [[ $# = 0 ]]; then
+        local MODE="all"
+    elif [[ "$1" == "all" ]]; then
+        local MODE="all"
+        shift
+    elif [[ "$1" == "rootfs" ]]; then
+        local MODE="rootfs"
+        shift
+    elif [[ "$1" == "kernels" ]]; then
+        local MODE="kernels"
+        shift
+    elif [[ "$1" == "help" ]]; then
+        print_help
+        exit 0
+    else
+        die "Unknown arg: $1. Please use \`$0 help\` for help."
+    fi
```

Comment:
```
case $1 in 
    all|rootfs|kernels)
        local MODE=$1
        shift
    help)
        print_help
        exit 0
        ;;
    *)
        die Unknown
esac
```

Reply:
but I would still need the `if [[ $# -eq 0 ]]` case, right?

Reply:
done